### ISSUE TITLE
Influence of object scale

### ISSUE THREAD

**Issue by MartinSmeyer** [NVlabs/nvdiffrecmc#12]
This is really awesome work, thank you!

I got good initial results for an image collection of an industrial object including both metallic and plastic parts:
![val_000359_kd](https://user-images.githubusercontent.com/18601778/201074361-6b916f2d-3c52-4074-9726-8c3d5f50ed5b.png) ![val_000359_ks](https://user-images.githubusercontent.com/18601778/201074369-83cd4e77-2935-4f8e-9b4d-01b45a8ef771.png) ![val_000359_light_image](https://user-images.githubusercontent.com/18601778/201074373-ce3e325d-72c7-4455-923d-67c2601f3d4e.png) ![val_000359_normal](https://user-images.githubusercontent.com/18601778/201074380-a75681a5-9596-47b0-8aa3-c5258a6707c4.png) ![val_000359_opt](https://user-images.githubusercontent.com/18601778/201074383-0411fed3-9b4f-4707-b974-d9feb2b4b9e5.png) ![val_000359_ref](https://user-images.githubusercontent.com/18601778/201074386-9ccb097d-d98b-4208-9341-8043b0241520.png)

Since some geometric detail is still hidden in the texture, I was wondering if the small scale (6cm diameter) might be a problem? I tried playing with the `mesh_scale` parameter but it diverges so I assume it only works for input meshes not for posed image collections?

```
{
    "ref_mesh": "data/tless_half/obj_000001",
    "random_textures": true,
    "iter": 5000,
    "save_interval": 100,
    "texture_res": [ 512, 512 ],
    "train_res": [280, 280],
    "batch": 32,
    "learning_rate": [0.06, 0.01],
    "dmtet_grid" : 128,
    "mesh_scale" : 1.0,
    "validate" : true,
    "n_samples" : 12,
    "laplace_scale" : 3000,
    "denoiser" : "bilateral",
    "display": [{"latlong" : true}, {"bsdf" : "kd"}, {"bsdf" : "ks"}, {"bsdf" : "normal"}],
    "background" : "black",
    "out_dir": "tless_half_obj_000001_bs_lr_tex512"
}
```

**Comment by MartinSmeyer** [comment]
I guess with a little finer subdivisions we could gain a lot:

![image](https://user-images.githubusercontent.com/18601778/201096640-4e70c64f-c510-47a8-b62b-abba1011136c.png)

**Comment by MartinSmeyer** [comment]
Actually my fault sorry, I used `mesh_scale=5` instead of `mesh_scale=1/5`.

**Comment by jmunkberg** [comment]
Thanks @MartinSmeyer 

Yes, the code is fairly rudimentary for posed image collections. 
For best results, make sure your object is centered, and that the grid is fairly tight around the object. 
There is a known issue with mesh_scale < 1, so if possible, it would be better to scale the mv matrix in the dataset loader if that is an option for you. Which loader are you using for this test?

**Comment by jmunkberg** [comment]
Also, if possible, I would recommend to run in higher image resolution than "train_res": [280, 280] to increase geometry detail.

**Comment by MartinSmeyer** [comment]
Thank you very much for your insights, @jmunkberg!

Yes, I think I am running into the issue for `mesh_scale<1` now:
![image](https://user-images.githubusercontent.com/18601778/201123397-e8f5a1a9-2a16-48f2-b7d1-817aafe5b0c8.png)

> Which loader are you using for this test?

I am using the `DatasetLLFF`. Could you please point me to the line where I can scale the mv matrix? 

> I would recommend to run in higher image resolution than "train_res": [280, 280] 

Good point, I have another set of higher resolution images which I will test next.

**Comment by jmunkberg** [comment]
After line 88 https://github.com/NVlabs/nvdiffrecmc/blob/main/dataset/dataset_llff.py#L88
try adding a line `mv   = mv @ util.scale(5.0)`  or `mv   = mv @ util.scale(1.0/5.0)`  (I don't remember which way to scale in the LLFF format)
Untested, but hopefully something like that can work!

That loader should already auto-center the poses as well
https://github.com/NVlabs/nvdiffrecmc/blob/main/dataset/dataset_llff.py#L65

**Comment by MartinSmeyer** [comment]
Yes thanks, I just figured it out, it looks promising:
![image](https://user-images.githubusercontent.com/18601778/201132763-d0fcdbaa-b414-4c3e-857e-0aec5e8af3da.png)

**Comment by MartinSmeyer** [comment]
I did `mv[:3,3] *= 5` so `mv = mv @ util.scale(5.0)`

**Comment by MartinSmeyer** [comment]
It's getting more details now:

Flat:
![image](https://user-images.githubusercontent.com/18601778/201156162-cb2ed5b3-9f46-4f53-bd3a-577942faf5c6.png)

Smooth:
![image](https://user-images.githubusercontent.com/18601778/201156353-e5139ec1-431f-4db0-9fb7-2cecb6d8f1a3.png)

The metal material part is unfortunately missing this time:
![image](https://user-images.githubusercontent.com/18601778/201156635-1e11698e-af9b-4e89-a328-25320f232c55.png)

Thanks for your help!

**Comment by jmunkberg** [comment]
Great! I would still expect much higher tessellation/triangle count, if you are using ""dmtet_grid" : 128," so ensure that the initial grid is fairly tight around your model (that is usually visible in early training images). Try scaling even more perhaps, like `mv = mv @ util.scale(10.0)` or higher.

**Comment by jmunkberg** [comment]
To get better material parameters, you can play a bit with the regularizers. The settings here are a good start if you are working from real photographs

https://github.com/NVlabs/nvdiffrecmc/blob/main/configs/nerd_gold.json

**Comment by MartinSmeyer** [comment]
Yes, I am using the 128 dmtet_grid, with 10 it already cuts off the top:
![image](https://user-images.githubusercontent.com/18601778/201172101-58c4318b-5550-4e9b-a448-e0b8c5b25b1a.png)

Currently trying 8, it looks better. For the material it seems like it still needs a higher loss on missing metallness despite the boundaries. Higher resolution should help, too, will report back.

**Comment by MartinSmeyer** [comment]
Auto smooth shading with mv scaled by 8:

![image](https://user-images.githubusercontent.com/18601778/201322518-997d1f82-01cc-4214-bcce-11342f0ccf0a.png)


Got a bit of the metallicness as well:
![image](https://user-images.githubusercontent.com/18601778/201323176-4b9621aa-a96d-40ae-bacf-abc50cb38e4e.png)

I still need to preprocess the high resolution data, will close this for now but probably report back results.

Thanks again!

**Comment by MartinSmeyer** [comment]
As an update, higher resolution data improved geometry, unfortunately the metallic parts are missing completely now although I tried playing around with the parameters. I might try changing the `lambda` parameters as well. Maybe it's because it was recorded on a turntable, without a constant point light source. Or because there are not so many metallic regions.

![tless_1](https://user-images.githubusercontent.com/18601778/201969855-a59cf499-910a-492f-a8ff-5d528d802cec.png)
![tless_1_textured](https://user-images.githubusercontent.com/18601778/201969859-9cb6084b-5aa0-47d7-ac44-b156d0ad0c4a.png)

Original:
![000414](https://user-images.githubusercontent.com/18601778/201969950-728f3b6c-da25-408a-b301-fbcfbf720ca4.png)

Thanks again!

**Comment by jmunkberg** [comment]
> Maybe it's because it was recorded on a turntable, without a constant point light source.

Yes, that is likely the issue. We assume a fixed lighting condition and a static object, 
so if the object rotates, that will break our assumption. For a turntable setup, we could likely add a transform to the env map for each frame, to take the rotation into account, but that is not supported atm
